DIGNDB.App.SmitteStop.API/Attributes/UploadKeysAuthorizationAttribute.cs

```diff
@@ -0,0 +1,82 @@
+﻿using DIGNDB.App.SmitteStop.Core.Contracts;
+using DIGNDB.App.SmitteStop.Domain.Configuration;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using DIGNDB.App.SmitteStop.API.Contracts;
+using AnonymousTokens.Server.Protocol;
+using AnonymousTokens.Core.Services.InMemory;
+using Org.BouncyCastle.Crypto.EC;
+using Org.BouncyCastle.Asn1.X9;
+using System.Security.Cryptography;
+
+namespace DIGNDB.App.SmitteStop.API.Attributes
+{
+    public class UploadKeysAuthorizationAttribute : ActionFilterAttribute
+    {
+        private readonly AppSettingsConfig _appSettingsConfig;
+        private readonly AuthOptions _authOptions;
+        private readonly IJwtValidationService _jwtValidationService;
+        private readonly ITokenVerifier _tokenVerifier = new TokenVerifier(new InMemorySeedStore());
+        private readonly InMemoryPrivateKeyStore _privateKeyStore = new InMemoryPrivateKeyStore();
+
+        public UploadKeysAuthorizationAttribute(AppSettingsConfig appSettingsConfig, AuthOptions authOptions, IJwtValidationService jwtValidationService)
+        {
+            _appSettingsConfig = appSettingsConfig;
+            _authOptions = authOptions;
+            _jwtValidationService = jwtValidationService;
+        }
+
+        public override async void OnActionExecuting(ActionExecutingContext context)
+        {
+            if (!_authOptions.AuthHeaderCheckEnabled) return;
+
+            var logger = context.HttpContext.RequestServices.GetService<ILogger<DiagnosticKeysController>>();
+            try
+            {
+                string authHeader = context.HttpContext.Request.Headers["Authorization"];
+                if (authHeader != null && authHeader.Contains("Bearer"))
+                {
+                    var token = authHeader.Replace("Bearer", string.Empty);
+                    token = token.Trim();
+
+                    _jwtValidationService.IsTokenValid(token);
+                    return;
+
+                }
+                else if (authHeader != null && authHeader.StartsWith("Anonymous ")) 
+                {
+                    var _ecParameters = CustomNamedCurves.GetByOid(X9ObjectIdentifiers.Prime256v1);
+                    string[] anonymousToken = authHeader.Replace("Anonymous ", string.Empty).Split(".");
+
+                    var W = _ecParameters.Curve.DecodePoint(Convert.FromBase64String(anonymousToken[0]));
+                    var t = Convert.FromBase64String(anonymousToken[1]);
```

Comment:
Why not give this `t` more meaningful name e.g. `tokenSeed`?

Reply:
Pending a decision from FHI, I'm tending to keeping the cryptographic variables as they are described in the cryptography, rather than making translations. The whole code section may be appropriate to extract into a new class (similar to JwtValidationService, but focus has to be on the app now.

Reply:
I will still argue about it.

Please take a look at documentation comments of `VerifyTokenAsync` method.
```
/// <summary>
/// Used by the token verifier. It recreates the initial point from the initiator, signs it, and verifies that they are equal.
/// </summary>
/// <param name="k">The private key for the token scheme</param>
/// <param name="curve">Curve parameters</param>
/// <param name="t">Seed for the initial point chosen by the initiator</param>
/// <param name="W">Token received from the initiator</param>
/// <returns>True if the token is valid, otherwise false</returns>
```

I'm know close to nothing when it comes to cryptography, but the method you're using for token verification `VerifyTokenAsync` clearly states that `t` is `Seed for the initial point chosen by the initiator`.
Why not call it `seed` and be compliant with the used library? 

Also looking at some random RFC ([RFC5639](https://tools.ietf.org/html/rfc5639)),
in part `Appendix A.  Pseudo-Random Generation of Parameters` it says:
```
For updating the seed s of 160-bit length we use the following
   function update_seed(s):
```
and seed is called `s` there.

I'd still suggest calling it with whole words.

Reply:
`tokenSeed` is a good suggestion.